tests/test_functions.py

```diff
@@ -38,37 +38,41 @@
 SLEEP_TIME = 3
 
 
-def check_sumo(case_uuid, tag_prefix, correct, class_type, sumo):
-    # There has been instances when this fails, probably because of
-    # some time delay, have introduced a little sleep to make it not fail
+def check_sumo(case_uuid, class_type, expected_count, sumo):
     sleep(SLEEP_TIME)
-    if not tag_prefix.endswith("*"):
-        tag_prefix = tag_prefix + "*"
 
-    path = f"/objects('{case_uuid}')/children"
-    query = f"$filter=data.tagname:{tag_prefix}"
-
-    if class_type != "*":
-        query += f" AND class:{class_type}"
-        check_nr = correct
-    else:
-        # Plus one because we always upload parameters.txt automatically
-        check_nr = correct + 1
-
-    results = sumo.get(path, query).json()
-
-    returned = results["hits"]["total"]["value"]
-    assert returned == check_nr, (
-        f"Supposed to upload {check_nr}, but actual were {returned}"
-    )
+    results = sumo.post(
+        "/search",
+        json={
+            "query": {
+                "bool": {
+                    "must": [
+                        {
+                            "term": {
+                                "_sumo.parent_object.keyword": {
+                                    "value": case_uuid
+                                }
+                            }
+                        },
+                        {"term": {"class.keyword": class_type}},
+                    ],
+                    "must_not": [
+                        {"term": {"_sumo.hidden": {"value": "true"}}}
+                    ],
+                }
+            },
+            "size": expected_count,
```

Comment:
Should probably add 
```
"track_total_hits": true,
"_source": false,
```
--- as we don't want any of the document properties, and we want an accurate document count (although that is probably not going to be an issue here, as I assume there will be less than 10000 hits.)